gateway/utils/syncman/syncman_artifact.go

```diff
@@ -0,0 +1,64 @@
+package syncman
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/spaceuptech/space-cloud/gateway/modules/auth"
+	"github.com/spaceuptech/space-cloud/gateway/utils"
+)
+
+func (s *Manager) HandleArtifactRequests(auth *auth.Module) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// http: Request.RequestURI can't be set in client requests.
+		// http://golang.org/src/pkg/net/http/client.go
+		r.Host = strings.Split(s.artifactAddr, ":")[0]
+		r.URL.Host = s.artifactAddr
+
```

Comment:
Check if user is authorised to make the request here using admin man. Also do the same for the handler forwarding all runner requests